packages/messaging/test/iid-model-get-token.test.ts

```diff
@@ -0,0 +1,148 @@
+/**
+ * Copyright 2017 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { expect } from 'chai';
+import * as sinon from 'sinon';
+import IIDModel from '../src/models/iid-model';
+import Errors from '../src/models/errors';
+import { fetchMock } from './testing-utils/mock-fetch';
+import makeFakeSubscription from './make-fake-subscription';
+
+describe('Firebase Messaging > IIDModel.getToken()', function() {
+  const fcmSenderId = '1234567';
+  const fcmToken = 'qwerty';
+  const fcmPushSet = '7654321';
+  const subscription = makeFakeSubscription();
+  const appPubKey = new Uint8Array([
+    255,
+    237,
+    107,
+    177,
+    171,
+    78,
+    84,
+    131,
+    221,
+    231,
+    87,
+    188,
+    22,
+    232,
+    71,
+    15
+  ]).buffer;
+
+  const sandbox = sinon.sandbox.create();
+  let stubedFetch;
+  let globalIIDModel;
+
+  const cleanUp = () => {
+    sandbox.restore();
+    globalIIDModel = null;
+  };
+
+  before(function() {
+    stubedFetch = sinon.stub(window, 'fetch');
```

Comment:
Please move this into `beforeEach()` after cleanup and use the sandbox.stub().

Generally, if you stub out a method in the global scopes, its trivial to slip up and cause all future tests to fail because a stub isn't cleared up properly.